src/app/shared/components/notifications/notifications-list/notifications-list.component.ts

```diff
@@ -1,311 +1,233 @@
+    if (notification.readDateTime) {
+      return;
     }
 
-    if (descriptionType === NotificationDescriptionType.Short) {
-      return this.getShortNotificationMessage(notificationType, action);
-    }
+    this.store.dispatch(new ReadUsersNotificationById(notification));
 
-    return Constants.NO_INFORMATION;
+    notification.readDateTime = new Date(Date.now());
+    this.notificationsAmount.amount--;
   }
 
-  private getDataStatusNotification(
-    descriptionType: NotificationDescriptionType,
-    notificationType: NotificationType,
-    status: ApplicationStatuses | ProviderStatuses
-  ): string {
-    switch (descriptionType) {
-      case NotificationDescriptionType.Full:
-        return this.getFullNotificationMessage(notificationType, status);
-      case NotificationDescriptionType.Short:
-        return this.getShortNotificationMessage(notificationType, status);
-      default:
-        return Constants.NO_INFORMATION;
-    }
+  public onDeleteAll(): void {
+    this.notificationsGroupedByType = [];
+    this.notifications = [];
+    this.onReadAll();
   }
 
-  private getDataLicenseNotification(descriptionType: NotificationDescriptionType, status: ApplicationStatuses | ProviderStatuses): string {
-    switch (descriptionType) {
-      case NotificationDescriptionType.Full:
-        return NotificationProviderLicenseFullDescription[status];
-      case NotificationDescriptionType.Short:
-        return NotificationProviderLicenseShortDescription[status];
-      default:
-        return Constants.NO_INFORMATION;
-    }
+  // TODO: Consider deleting because of unused
+  public onDeleteSingle(notification: Notification): void {
+    this.store.dispatch(new DeleteUsersNotificationById(notification.id));
+
+    this.notificationsAmount.amount--;
+    this.notifications = this.notifications.filter((receivedNotification: Notification) => receivedNotification.id !== notification.id);
   }
 
-  private addRecievedNotification(recievedNotification: Notification): void {
+  private addReceivedNotification(receivedNotification: Notification): void {
     this.notificationsAmount.amount++;
 
-    if (recievedNotification.type !== NotificationType.Application && recievedNotification.type !== NotificationType.Chat) {
-      this.notifications.push(recievedNotification);
+    if (receivedNotification.type !== NotificationType.Application && receivedNotification.type !== NotificationType.Chat) {
+      this.notifications.unshift(receivedNotification);
       return;
     }
 
-    for (const groupByType of this.groupsByType) {
-      if (groupByType.type === recievedNotification.type) {
-        for (const groupByAction of groupByType.groupedByAdditionalData) {
-          if (groupByAction.groupedData === recievedNotification.data.Status) {
-            groupByType.amount++;
-            groupByAction.amount++;
-
+    for (const notificationGroupedByType of this.notificationsGroupedByType) {
```

Comment:
It is better to use internal methods for iterations rather than for... loops, as they have better performance